repl/src/main/scala/com/cloudera/livy/repl/Session.scala

```diff
@@ -67,9 +69,9 @@ class Session(interpreter: Interpreter, stateChangedCallback: SessionState => Un
       changeState(SessionState.Idle())
       sc
     }
-    future.onFailure { case _ =>
-      changeState(SessionState.Error())
-    }
+
+    future.onFailure { case _ => changeState(SessionState.Error()) }
+    future.onSuccess { case sc => _sc = Option(sc).orElse(Some(SparkContext.getOrCreate())) }
```

Comment:
I think we can call SparkContext.getOrCreate() here, https://github.com/cloudera/livy/blob/master/repl/src/main/scala/com/cloudera/livy/repl/ProcessInterpreter.scala#L56, so that we can always get active SparkContext here. Otherwise something wrong must happens.